src/db/commit.rs

```diff
@@ -183,7 +220,15 @@ impl Commit {
         let commit = commit_fb::Commit::create(&mut builder, commit_args);
         builder.finish(commit, None);
 
-        let chunk = dag::Chunk::new(builder.collapse(), &[value_hash]);
+        let mut refs = vec![value_hash];
+        if let Ref::Strong(basis_hash) = basis_hash {
+            refs.push(basis_hash);
+        }
+        if let Some(original_hash) = original_hash {
```

Comment:
I think we have to change this to a weak reference when sync is complete and remove the original from refs. But now that I think about it, that would actually change the hash. If original were a str instead of a typed thing then 'making it a weak ref' is just removing it from the refs -- the commit is untouched.

Reply:
Once again I might be mistaken but based on looking at the code...

> If original were a str instead of a typed thing

We do not have a typed thing for refs. We have a string stored in the `data` and it is up to the person creating the chunk to also include the hash in the `refs` that are strong.

Changing the refs does not change the hash.

Reply:
Gah, my bad. I overlooked that the notion of strong or weak isn't represented at the commit fb level. I guess because how we've used the dag historically I find that surprising, like if we wanted to verify that commit chains are well-formed we'd have to use our historical knowledge of the semantics of each field containing a hash at the time the commit was written, instead of ignoring weak refs and checking that all strong refs exist. But in the constrained way we are using commits in repc I guess it makes sense. 

I also forgot that chunk hash is just over the data and not data+refs. This is both a cool feature because we can delete refs under the hood when they are no longer needed and a little scary that we might inadvertently overwrite write a commit with the same data but different refs. Like if at any time in two places we want to create a strong ref -- say in order to make the named head a strong ref -- we might be tempted to use a chunk with no data but set its refs. However *all* chunks with empty data share the same hash even if they have different refs, so we could easily clobber ourselves. As another example suppose we chunked the contents in the value map we reference in each commit in different ways. Then we might end up with two chunks holding the same data and thus with the same hash, but with different refs. 

Anyway, I think you are clear on the underlying thing I was trying to say, which is that when we finalize sync for each replayed local commit we need to remove its original hash from its chunk's refs. Yes?

I will take my answer to 'how are we going to make named heads strong refs' off the air as it is not in scope for your set of changes here.